packages/ui/Marquee/MarqueeDecorator.js

```diff
@@ -478,42 +498,46 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 
 			// TODO: absolute showing check (or assume that it won't be rendered if it isn't showing?)
 			if (node && this.distance == null && !this.props.marqueeDisabled) {
-				this.distance = this.calculateDistance(node);
+				const {width} = node.getBoundingClientRect();
+				const {scrollWidth} = node;
+
+				this.padding = this.getPadding(width);
+				this.distance = this.calculateDistance(width, scrollWidth, this.padding);
 				this.contentFits = !this.shouldAnimate(this.distance);
+			}
+		}
 
-				const overflow = this.calculateTextOverflow(this.distance);
-				this.setState(state => {
-					return state.overflow === overflow ? null : {overflow};
-				});
+		getPadding (width) {
+			const {marqueePadding} = this.props;
+
+			if (typeof marqueePadding === 'string') {
+				if (/^\d+(.\d+)?%$/.test(marqueePadding)) {
```

Comment:
Is this `/^\d+(.\d+)?%$/` supposed to be `/^\d+(\.\d+)?%$/` instead?
Start of string; 1 or more decimal characters; start capture; any single character; 1 or more decimal characters; end capture; flag the capture as optional; literal `%` character; end of string.

Reply:
Indeed!